vassal-app/src/main/java/VASSAL/build/module/map/CounterDetailViewer.java

```diff
@@ -722,15 +730,17 @@ else if (displayWhat.equals(EXC_LAYERS)) {
     protected int insertPos = 0;
     protected Point foundPieceAt;
     protected boolean showingOverlap;
+    protected CounterDetailViewer cdv;
 
-    public Visitor(Filter filter, Map map, Point pt, boolean showOverlap) {
+    public Visitor(Filter filter, Map map, Point pt, boolean showOverlap, CounterDetailViewer counterDetailViewer) {
```

Comment:
The `Visitor` doesn't need the CDG, just the values of two of its fields---and it needs those values only at construction time, not any time later. So, rather than break binary compatibility by changing `Visitor`'s only ctor, we can do the following:

* Add two `protected final boolean` fields to `Visitor`, for `showNumberFromDeck` and `showDeckMasked`.
* Deprecate the old ctor, and in it set those two fields to whatever the defaults in the CDV are for them.
* Make a new ctor that takes values for those two fields as arguments, and use that ctor below in `getDisplayablePieces()`.
* Remove the clirr exception for the `Visitor` ctor, because we don't need it anymore.

Reply:
Okay pushed that (hopefully)